### DISCUSSION TITLE
How to generate String key ?

### DISCUSSION THREAD

**Discussion by arindamAtlyikDotcom** [tekartik/sembast.dart#368] [Q&A]
As per the documentation String key is also supported and can be generated just like firestore. However I could not find any reference regarding that.
Is there any inbuilt way to generate **String** key instead of int key ?

**Comment by alextekartik** [comment]
Indeed String key generation is copied from firestore and is used when you call `store.add()`. You can also call `generateStringKey()` to generate a string key using the following (public) import:

```
import 'package:sembast/utils/key_utils.dart';
```
